pkg/splunk/enterprise/clustermanager.go

```diff
@@ -434,3 +442,95 @@ func VerifyCMisMultisite(ctx context.Context, cr *enterpriseApi.ClusterManager,
 	}
 	return extraEnv, err
 }
+
+// isClusterManagerReadyForUpgrade checks if it is suitable to update the clusterManager based on the Status of the licenseManager, returns bool, err accordingly
+func isClusterManagerReadyForUpgrade(ctx context.Context, c splcommon.ControllerClient, cr *enterpriseApi.ClusterManager) (bool, error) {
+
+	reqLogger := log.FromContext(ctx)
+	scopedLog := reqLogger.WithName("isClusterManagerReadyForUpgrade").WithValues("name", cr.GetName(), "namespace", cr.GetNamespace())
+	eventPublisher, _ := newK8EventPublisher(c, cr)
+
+	namespacedName := types.NamespacedName{
+		Namespace: cr.GetNamespace(),
+		Name:      GetSplunkStatefulsetName(SplunkClusterManager, cr.GetName()),
+	}
+
+	// check if the stateful set is created at this instance
+	statefulSet := &appsv1.StatefulSet{}
+	err := c.Get(ctx, namespacedName, statefulSet)
+	if err != nil && k8serrors.IsNotFound(err) {
+		return true, nil
+	}
+
+	licenseManagerRef := cr.Spec.LicenseManagerRef
+	if licenseManagerRef.Name == "" {
+		return true, nil
+	}
+
+	namespacedName = types.NamespacedName{Namespace: cr.GetNamespace(), Name: licenseManagerRef.Name}
+
+	// create new object
+	licenseManager := &enterpriseApi.LicenseManager{}
+
+	// get the license manager referred in cluster manager
+	err = c.Get(ctx, namespacedName, licenseManager)
+	if err != nil {
+		eventPublisher.Warning(ctx, "isClusterManagerReadyForUpgrade", fmt.Sprintf("Could not find the License Manager. Reason %v", err))
+		scopedLog.Error(err, "Unable to get licenseManager")
+		return true, err
+	}
+
+	cmImage, err := getCurrentImage(ctx, c, cr, SplunkClusterManager)
+	if err != nil {
+		eventPublisher.Warning(ctx, "isClusterManagerReadyForUpgrade", fmt.Sprintf("Could not get the Cluster Manager Image. Reason %v", err))
+		scopedLog.Error(err, "Unable to get clusterManager current image")
+		return false, err
+	}
+
+	// check conditions for upgrade
+	annotations := cr.GetAnnotations()
+	if annotations == nil {
+		annotations = map[string]string{}
+	}
+	if _, ok := annotations["splunk/image-tag"]; ok {
+		if (cr.Spec.Image != cmImage) && (licenseManager.Status.Phase != enterpriseApi.PhaseReady || licenseManager.Spec.Image != annotations["splunk/image-tag"]) {
```

Comment:
nit: Can we add a comment here. Something like:

Determining if there is an upgrade is going on & whether clusterManager is ready for the upgrade